### DISCUSSION TITLE
UART1 and UART2

### DISCUSSION THREAD

**Discussion by workpage2** [stm32duino/Arduino_Core_STM32#1561] [Q&A]
I am using a Blackpill 411CE board. I use generic F411CEUx in chip configuration.
To configure uart, I use the command:
`Serial.begin(115200);`
and
`Serial2.begin(115200);`
However, data exchange in both cases goes through PA2 and PA3.
How do I switch to PA9 and PA10?

**Comment by fpistm** [comment]
Hi @workpage2 
First, please use  the forum or the GH discussion to request support.

Generic F411CE define the Serial2 with `PA2`/`PA3` by default, `Serial` is mapped on `Serial2`.
https://github.com/stm32duino/Arduino_Core_STM32/blob/975da6ce61336b7df49ed88c31e3cd9c2539cf57/variants/STM32F4xx/F411C(C-E)(U-Y)/variant_generic.h#L128-L140

If you want to change that you can change the define using build_opt.h feature:
https://github.com/stm32duino/wiki/wiki/Customize-build-options-using-build_opt.h

or you can simply change the default pins to use `PA10`/PA9 as explained in the Wiki:
https://github.com/stm32duino/wiki/wiki/API#hardwareserial

```c++
    Serial.setRx(PA10);
    Serial.setTx(PA9);
    Serial.begin(9600);
```

Note those pins are linked to `USART1` not `USART2` like `PA2`/`PA3`.
